### PR TITLE
Refactor class names

### PR DESCRIPTION
**Pull request by marcustw**
Refactor class names

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -25,7 +25,7 @@ title: Developer Guide
   * [Glossary](#glossary)
 * [**Appendix: Instructions for manual testing**](#appendix_manual_testing)
   * [Launch and shutdown](#launch_shutdown)
-  * [Deleting a person](#deleting_a_person)
+  * [Deleting a student](#deleting_a_studentn)
```
Comment (nweiyue) [inline docs/DeveloperGuide.md]: Should "studentn" be "student instead?

File: docs/tutorials/AddRemark.md [hunk 2]
```diff
@@ -5,7 +5,7 @@ title: "Tutorial: Adding a command"
 
 Let's walk you through the implementation of a new command — `remark`.
 
-This command allows users of the AddressBook application to add optional remarks to people in their address book and edit it if required. The command should have the following format:
+This command allows users of the StudentList application to add optional remarks to people in their student list and edit it if required. The command should have the following format:
```
Comment (nweiyue) [inline docs/tutorials/AddRemark.md:8]: Should people be "student' instead?

File: docs/tutorials/AddRemark.md [hunk 3]
```diff
@@ -322,41 +322,41 @@ After the previous step, we notice a peculiar regression — we went from di
...
-        Person editedPerson = new Person(personToEdit.getName(), personToEdit.getPhone(), personToEdit.getEmail(),
-                personToEdit.getAddress(), remark, personToEdit.getTags());
+        Student studentToEdit = lastShownList.get(index.getZeroBased());
+        Student editedStudent = new Student(studentToEdit.getName(), studentToEdit.getPhone(), studentToEdit.getEmail(),
+                studentToEdit.getAddress(), remark, studentToEdit.getTags());
 
-        model.setPerson(personToEdit, editedPerson);
-        model.updateFilteredPersonList(PREDICATE_SHOW_ALL_PERSONS);
+        model.setStudent(studentToEdit, editedStudent);
+        model.updateFilteredStudentList(PREDICATE_SHOW_ALL_PERSONS);
```
Comment (nweiyue) [inline docs/tutorials/AddRemark.md]: Should the constant messages be renamed as well? I've noticed this issue in other places as well.

File: src/main/java/atas/storage/AtasStorage.java [hunk 4]
```diff
@@ -5,42 +5,42 @@
...
 
     /**
      * Returns the file path of the data file.
      */
-    Path getAddressBookFilePath();
+    Path getStudentListFilePath();
 
     /**
-     * Returns AddressBook data as a {@link ReadOnlyAddressBook}.
+     * Returns AddressBook data as a {@link ReadOnlyStudentList}.
```
Comment (nweiyue) [inline src/main/java/atas/storage/AtasStorage.java]: Should "AddressBook" be refactored to "StudentList"?

File: src/main/java/atas/storage/AtasStorage.java [hunk 5]
```diff
@@ -5,42 +5,42 @@
...
+    Optional<ReadOnlyStudentList> readStudentList(Path filePath) throws DataConversionException, IOException;
 
     /**
-     * Saves the given {@link ReadOnlyAddressBook} to the storage.
+     * Saves the given {@link ReadOnlyStudentList} to the storage.
      * @param addressBook cannot be null.
      * @throws IOException if there was any problem writing to the file.
      */
-    void saveAddressBook(ReadOnlyAddressBook addressBook) throws IOException;
+    void saveStudentList(ReadOnlyStudentList addressBook) throws IOException;
```
Comment (nweiyue) [inline src/main/java/atas/storage/AtasStorage.java]: Should the parameter be change to studentList instead?

File: src/main/java/atas/storage/JsonAtasStorage.java [hunk 6]
```diff
@@ -13,41 +13,41 @@
 import atas.commons.exceptions.DataConversionException;
 import atas.commons.exceptions.IllegalValueException;
 import atas.commons.util.JsonUtil;
-import atas.model.ReadOnlyAddressBook;
+import atas.model.ReadOnlyStudentList;
 
 /**
  * A class to access AddressBook data stored as a json file on the hard disk.
```
Comment (nweiyue) [inline src/main/java/atas/storage/JsonAtasStorage.java]: Should this be change to ATAS instead?

File: src/main/java/atas/storage/StorageManager.java [hunk 7]
```diff
@@ -19,18 +19,18 @@
 
     private static final Logger logger = LogsCenter.getLogger(StorageManager.class);
     private SessionListStorage sessionListStorage;
-    private AddressBookStorage addressBookStorage;
+    private AtasStorage atasStorage;
     private UserPrefsStorage userPrefsStorage;
 
     /**
      * Creates a {@code StorageManager} with the given {@code SessionListStorage}, {@code AddressBookStorage}
```
Comment (nweiyue) [inline src/main/java/atas/storage/StorageManager.java]: Should "AddressBookStorage" be change to "atasStorage" instead?

**Review by nweiyue — APPROVED** [review]
LGTM! Just some nits.

File: src/main/java/atas/storage/AtasStorage.java (hunk 4 above)
Comment (marcustw) [reply src/main/java/atas/storage/AtasStorage.java]: Thank you, I've missed it out!

File: docs/diagrams/ArchitectureSequenceDiagram.puml [hunk 8]
```diff
@@ -19,7 +19,7 @@ activate model MODEL_COLOR
 model -[MODEL_COLOR]-> logic
 deactivate model
 
-logic -[LOGIC_COLOR]> storage : saveAddressBook(addressBook)
+logic -[LOGIC_COLOR]> storage : saveAddressBook(studentList)
```
Comment (erisjacey) [inline docs/diagrams/ArchitectureSequenceDiagram.puml:22]: Should this be something like `saveStudentList`?

File: docs/tutorials/TracingCode.md [hunk 9]
```diff
@@ -180,7 +180,7 @@ Recall from the User Guide that the `edit` command has the format: `edit INDEX [
     * {@code JsonSerializableAddressBook}.
     */
    public JsonSerializableAddressBook(ReadOnlyAddressBook source) {
```
Comment (erisjacey) [inline docs/tutorials/TracingCode.md:182]: Maybe change this to `JsonSerializableStudentList` and correspondingly `ReadOnlyStudentList`?

File: src/main/java/atas/model/UserPrefs.java [hunk 10]
```diff
@@ -49,11 +49,11 @@ public void setGuiSettings(GuiSettings guiSettings) {
...
 
-    public Path getAddressBookFilePath() {
+    public Path getStudentListFilePath() {
         return addressBookFilePath;
     }
 
-    public void setAddressBookFilePath(Path addressBookFilePath) {
+    public void setStudentListFilePath(Path addressBookFilePath) {
         requireNonNull(addressBookFilePath);
         this.addressBookFilePath = addressBookFilePath;
```
Comment (erisjacey) [inline src/main/java/atas/model/UserPrefs.java:58]: `addressBook` -> `studentList`?

File: src/main/java/atas/storage/StorageManager.java [hunk 11]
```diff
@@ -55,30 +55,30 @@ public void saveUserPrefs(ReadOnlyUserPrefs userPrefs) throws IOException {
     // ================ AddressBook methods ==============================
```
Comment (erisjacey) [inline src/main/java/atas/storage/StorageManager.java]: AddressBook -> StudentList?

File: src/test/java/atas/logic/LogicManagerTest.java [hunk 12]
```diff
@@ -92,47 +92,47 @@ public void execute_validCommand_success() throws Exception {
...
     @Test
     public void getters() {
         JsonSessionListStorage jsonSessionListStorage =
                 new JsonSessionListStorage(temporaryFolder.resolve("typicalSessionSessionList.json"));
-        JsonAddressBookStorage addressBookStorage =
-                new JsonAddressBookStorage(temporaryFolder.resolve("addressBook.json"));
+        JsonAtasStorage addressBookStorage =
+                new JsonAtasStorage(temporaryFolder.resolve("addressBook.json"));
         JsonUserPrefsStorage userPrefsStorage = new JsonUserPrefsStorage(temporaryFolder.resolve("userPrefs.json"));
         StorageManager storage = new StorageManager(jsonSessionListStorage, addressBookStorage, userPrefsStorage);
```
Comment (erisjacey) [inline src/test/java/atas/logic/LogicManagerTest.java]: addressBook -> studentList?

File: src/main/java/atas/model/UserPrefs.java (hunk 10 above)
Comment (marcustw) [reply src/main/java/atas/model/UserPrefs.java:58]: Thank you for pointing it out!

**Review by erisjacey — APPROVED** [review]
LGTM!

**Pull request merged**